cypress/integration/checkout/checkout.basic.spec.js

```diff
@@ -1,54 +1,122 @@
     cy.getAdminToken()
       .then(() => {
-        cy.getLoanTypes({ limit: 1 });
-        cy.getMaterialTypes({ limit: 1 });
-        cy.getLocations({ limit: 1 });
-        cy.getHoldingTypes({ limit: 1 });
-        source = InventoryHoldings.getHoldingSources({ limit: 1 });
-        cy.getInstanceTypes({ limit: 1 });
-        cy.getUsers({ limit: 1, query: '"personal.firstName"="checkin-all" and "active"="true"' });
+        ServicePoints.createViaApi(servicePoint);
+        defaultLocation = Location.getDefaultLocation(servicePoint.id);
+        Location.createViaApi(defaultLocation);
+        cy.getInstanceTypes({ limit: 1 }).then((instanceTypes) => {
+          itemData.instanceTypeId = instanceTypes[0].id;
+        });
+        cy.getHoldingTypes({ limit: 1 }).then((res) => {
+          itemData.holdingTypeId = res[0].id;
+        });
+        cy.getLoanTypes({ limit: 1 }).then((res) => {
+          itemData.loanTypeId = res[0].id;
+        });
+        cy.getMaterialTypes({ limit: 1 }).then((res) => {
+          itemData.materialTypeId = res.id;
+          itemData.materialTypeName = res.name;
+        });
       })
       .then(() => {
-        cy.getUserServicePoints(Cypress.env('users')[0].id);
-
-        cy.createInstance({
+        InventoryInstances.createFolioInstanceViaApi({
           instance: {
-            instanceTypeId: Cypress.env('instanceTypes')[0].id,
-            title: `Pre-checkout instance ${Number(new Date())}`,
+            instanceTypeId: itemData.instanceTypeId,
+            title: itemData.instanceTitle,
           },
-          holdings: [{
-            holdingsTypeId: Cypress.env('holdingsTypes')[0].id,
-            permanentLocationId: Cypress.env('locations')[0].id,
-            sourceId: source.id,
-          }],
+          holdings: [
+            {
+              holdingsTypeId: itemData.holdingTypeId,
+              permanentLocationId: defaultLocation.id,
+            },
+          ],
           items: [
-            [{
-              barcode: ITEM_BARCODE,
-              missingPieces: '3',
-              numberOfMissingPieces: '3',
+            {
+              barcode: itemData.barcode,
               status: { name: 'Available' },
-              permanentLoanType: { id: Cypress.env('loanTypes')[0].id },
-              materialType: { id: Cypress.env('materialTypes')[0].id },
-            }],
+              permanentLoanType: { id: itemData.loanTypeId },
+              materialType: { id: itemData.materialTypeId },
+            },
           ],
         });
+      })
+      .then((specialInstanceIds) => {
+        itemData.testInstanceIds = specialInstanceIds;
       });
-  });
 
-  it('Basic flow', function () {
-    cy.visit('/checkout');
+    cy.createTempUser(
+      [
+        permissions.checkinAll.gui,
+        permissions.uiCirculationViewCreateEditDelete.gui,
+        permissions.uiUserCreate.gui,
+        permissions.inventoryAll.gui,
+        permissions.checkoutCirculatingItems.gui,
```

Comment:
I don't see here the permission Settings (Circ): Can create, edit and remove other settings

Reply:
We have such precondition. Please add it to your test
Setting: Circulation > Other settings, Username checkbox = true

Reply:
I don't see here the permission *Users: Can create new user;